### DISCUSSION TITLE
.update() and changing synthetic image brightness

### DISCUSSION THREAD

**Discussion by yahelleidenuniv** [DeepTrackAI/DeepTrack2#75] [Q&A]
Hi All, 

Hopefully a quick question; 

First of all is how to change the brightness of a synthetic image (i.e optics(scatterer) ). I try to do something like illumination  = dt.Add(value = 5) , then include this in the optics section. However this is very hard to tell whether it does anything, obviously there is the illumination gradient, but I just need to change the whole brightness of the image to make the histogram of my data and the synthetic data to match. 

To add to this; deeptrack seems to have some problems not updating whenever I do something like "image_of_particle.update()" i.e if I want to check if the illumination changes anything.  Nothing of the updated image changes (even positions of the particles if i choose the position to be random within a range). I use multiple scatterers in one sample.

**Comment by BenjaminMidtvedt** [comment]
Hi! Sorry for the late reply, this one slipped through the cracks.

Keep in mind that typical plotting functions (such as matplotlib) will automatically normalize the images, meaning that the apparent brightness can stay the same, even if you are correctly increasing the brightness of the image. If you are comparing against a experimental reference, I recommend setting `vmin` and `vmax` in `imshow` to ensure you get the same normalization.

Regarding updating, it's hard to diagnose without the code, but I'd first make sure that the position argument is a lambda function (position=lambda: np.random.rand(2)). If it is not a lambda, deeptrack has no way to receive a new set of positions for your scatterers!

Best of luck
